From 54fe7000bec5c124e084fa4596f69a14d45f2878 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 21 Dec 2017 16:40:10 +0200
Subject: [PATCH 2204/2241] fix: net: mvpp2x: set err as int instead of u32 in
 platform data get

Issue:
In mv_pp2x_platform_data_get procedure err variable were declared as u32
instead of int. clk_prepare_enable and mv_pp22_uio_mem_map return
negative in case of failure, in this cause if statements in
mv_pp2x_platform_data_get procedure won't recognize failure.

Fix:
Set err as int instead of u32 in mv_pp2x_platform_data_get procedure.

Change-Id: I8dc8fb4ed01fd083c89b4b4242b4949345759940
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48017
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: Yan Markman <ymarkman@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 5428fce..45e752d 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5762,7 +5762,7 @@ static int mv_pp2x_platform_data_get(struct platform_device *pdev,
 	struct device_node *dn = pdev->dev.of_node;
 	struct resource *res;
 	resource_size_t mspg_base, mspg_end;
-	u32	err;
+	int	err;
 
 	match = of_match_node(mv_pp2x_match_tbl, dn);
 	if (!match)
-- 
2.7.4

